### PR TITLE
added some examples to prod users and integrations + linked README

### PR DESCRIPTION
**Pull request by pgburt**
Added a few examples to the new docs. Also attempting to standardize how we link to those docs, from the README.

cc @ericchiang @philips

### CODE REVIEW

File: Documentation/production-users.md [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
 
-This document tracks people and use cases for dex in production. [Join the community](https://github.com/coreos/dex/), and help us keep the list up-to-date.
\ No newline at end of file
+This document tracks people and use cases for dex in production. [Join the community](https://github.com/coreos/dex/), and help us keep the list up-to-date.
+
+## [Tectonic](https://coreos.com/tectonic/)
+The login for Tectonic (a 100% pure upstream distribution of Kubernetes) pairs dex along with Kubernetes' RBAC.
+
+## [Quay.io](https://quay.io/)
+Quay.io is an enterprise grade container registry. It relies on dex to simplify user authentication.
```
Comment (ericchiang) [inline Documentation/production-users.md]: Quay currently doesn't use dex.

Comment (pgburt) [inline Documentation/production-users.md]: Day 57, my captors still haven't recognized my true identity as a lumpy potato. 

(woops, the Quay repo returns some false positives when searching for dex D: I'll fix this, and ask the team about false pos on Monday)

**Comment by ericchiang** [comment]
bump @pgburt

**Comment by ericchiang** [comment]
Closing since this hasn't been acted on in a while. Feel free to rebase and re-submit.

**Pull request closed without merging**